### DISCUSSION TITLE
Reduce lines and duplicate of codes

### DISCUSSION THREAD

**Discussion by faca5** [opencart/opencart#13989] [Ideas]
Hello.

Reduce lines of code with removing duplicates lines.

Example:
Filter can have own library to generate parameters like you solve with "language".
OpenCart v2 had for each translate own line in PHP file which make a lot of lines in source code.
OpenCart v3 had new system which load automatically language files from language file and reduce number of lines in PHP code.
OpenCart v4 should have similar system for all other DUPLICATES lines like FILTER, This could reduce for another 20% of file and improve performance of code.

Example of duplicate lines in PHP file:
admin/controller/catalog/product.php

```
		if (isset($this->request->get['filter_name'])) {
			$filter_name = $this->request->get['filter_name'];
		} else {
			$filter_name = '';
		}

		if (isset($this->request->get['filter_model'])) {
			$filter_model = $this->request->get['filter_model'];
		} else {
			$filter_model = '';
		}

		if (isset($this->request->get['filter_price'])) {
			$filter_price = $this->request->get['filter_price'];
		} else {
			$filter_price = '';
		}

		if (isset($this->request->get['filter_quantity'])) {
			$filter_quantity = $this->request->get['filter_quantity'];
		} else {
			$filter_quantity = '';
		}

		if (isset($this->request->get['filter_status'])) {
			$filter_status = $this->request->get['filter_status'];
		} else {
			$filter_status = '';
		}
```

**Comment by osworx** [comment]
There are more repeated code which could be simplified and make the whole core smaller!
